File: jdroid-android-core/src/debug/java/com/jdroid/android/debug/PreferencesAppender.java
package com.jdroid.android.debug;

import android.preference.PreferenceGroup;

import com.jdroid.java.collections.Lists;

import java.io.Serializable;
import java.util.List;

import androidx.appcompat.app.AppCompatActivity;

public abstract class PreferencesAppender implements Serializable {

	public abstract int getNameResId();

	public abstract void initPreferences(AppCompatActivity activity, PreferenceGroup preferenceGroup);

	public Boolean isEnabled() {
		return true;
	}

	public List<String> getRequiredPermissions() {
		return Lists.newArrayList();
	}

	public String getSharedPreferencesName() {
		return null;
	}
}

File: jdroid-android-core/src/debug/java/com/jdroid/android/debug/PreferencesAppender.kt
package com.jdroid.android.debug

import android.preference.PreferenceGroup
import androidx.appcompat.app.AppCompatActivity
import java.io.Serializable

abstract class PreferencesAppender : Serializable {

    abstract fun getNameResId(): Int

    abstract fun initPreferences(activity: AppCompatActivity, preferenceGroup: PreferenceGroup)

    open fun isEnabled(): Boolean {
        return true
    }

    open fun getRequiredPermissions(): List<String> {
        return listOf()
    }

    open fun getSharedPreferencesName(): String? {
        return null
    }
}
